### PR TITLE
Product definition for Sentinel 2 ARD granules

### PR DESCRIPTION
**Pull request by harshurampur**

### CODE REVIEW

**Comment by mpaget** [comment]
I suggest we check with the @AMA-AC crew to clarify that this is compatible with their work on the UK-based S2 ARD definition. Such a check a doesn't have to be a blocker for reviewing this particular contribution, but its worth checking with others in the ODC community who are also working in this space. I'll forward the pull request email to them and perhaps they can have an input.

Is this S2 ARD definition built for the Aust Copernicus Data Hub version of S2 data? If so, and if these data differ from the UK-based version (AMA can tell us), then perhaps we need to be explicit about the source or relevance of this S2 ARD versus other variants.

**Comment by harshurampur** [comment]
@ mpaget I will check with Simon Oliver whether this product definition is just built for Alpha2 or can be compatible with other variants.

**Comment by harshurampur** [comment]
@mpaget I have changed the file name to be 'ga_ard.yaml'  as this is supported for just GA nbar, nbart and pq products and will not support the Aust Copernicus Data Hub version

File: docs/config_samples/dataset_types/ga_ard.yaml [hunk 1]
```diff
@@ -0,0 +1,495 @@
...
+
+- name: 't_contiguity'
```
Comment (andrewdhicks) [inline docs/config_samples/dataset_types/ga_ard.yaml]: How do the `t_` bands differ from the others?
They both have the same aliases, which could cause an issue if we use them for band lookup.

File: docs/config_samples/dataset_types/ga_ard.yaml [hunk 2]
```diff
@@ -0,0 +1,495 @@
...
+      bits: [0,1,2,3,4,5,6,7]
+      description: Pixel contiguity in band stack
+      values:
+        0: not_contiguous
+        1: contiguous
+- name: 'aerosol'
+  aliases: [band_01, B01, Band1]
+  units: '1'
+  dtype: uint16
+  nodata: -999
```
Comment (andrewdhicks) [inline docs/config_samples/dataset_types/ga_ard.yaml]: The `uint16` datatype can't support signed values such as -999.

File: docs/config_samples/dataset_types/ga_ard.yaml [hunk 3]
```diff
@@ -0,0 +1,495 @@
...
+
+- name: 't_contiguity'
+  units: '1'
+  dtype: uint8
+  nodata: 255
+  flags_definition: 
+    cfmask:
```
Comment (andrewdhicks) [inline docs/config_samples/dataset_types/ga_ard.yaml]: Should be `contiguity`?

File: docs/config_samples/dataset_types/ga_ard.yaml [hunk 4]
```diff
@@ -0,0 +1,495 @@
...
```
Comment (andrewdhicks) [inline docs/config_samples/dataset_types/ga_ard.yaml]: Should be `contiguity`?

File: docs/config_samples/dataset_types/ga_ard.yaml [hunk 5]
```diff
@@ -0,0 +1,495 @@
...
+        2: shadow
+        3: snow
+        4: cloud
+
+- name: 'contiguity'
+  units: '1'
+  dtype: uint8
+  nodata: 255
+  flags_definition: 
+    cfmask:
```
Comment (andrewdhicks) [inline docs/config_samples/dataset_types/ga_ard.yaml]: Should be `contiguity`?

File: docs/config_samples/dataset_types/ga_ard.yaml [hunk 6]
```diff
@@ -0,0 +1,495 @@
...
+      description: Pixel contiguity in band stack
+      values:
+        0: not_contiguous
+        1: contiguous
+- name: 'aerosol'
+  aliases: [band_01, B01, Band1]
+  units: '1'
+  dtype: uint16
+  nodata: -999
+  spectral_definition:
```
Comment (andrewdhicks) [inline docs/config_samples/dataset_types/ga_ard.yaml]: Do 2A and 2B have different spectral response functions?

**Review by andrewdhicks — CHANGES_REQUESTED** [review]

File: docs/config_samples/dataset_types/ga_ard.yaml (hunk 2 above)
Comment (harshurampur) [reply docs/config_samples/dataset_types/ga_ard.yaml]: Yeah that's right, should have been 0. This was written by Simon, I am adding to the repo on his behalf I will confirm about the changes again. 
Thanks Andrew

File: docs/config_samples/dataset_types/ga_ard.yaml (hunk 1 above)
Comment (harshurampur) [reply docs/config_samples/dataset_types/ga_ard.yaml]: It differs as with prefix t_ as  nbar_t specifications and without as nbar.

**Review by harshurampur — COMMENTED** [review]
Updates according to the comments above : 
1. File name changed to ga_s2_ard.yaml to keep it specific to GA NBAR NBART  products 
2. Updated t_ as prefix to NBART data aliases within ARD product 
3. Confirm S2A and S2B have differing spectral responses (spectral response is measured on instrument prior to launch)
4. Contiguity replaces FMask as noted
5. uint16 updated to int16

**Review by andrewdhicks — APPROVED** [review]

**Pull request merged**